src/screens/MyPromotions.tsx

```diff
@@ -150,25 +150,86 @@ export default function MyPromotions(): ReactElement {
    */
   const onSaveButtonClick = useCallback(
     (promotionId: string) => {
-      if (!authUser) {
-        return Promise.reject(new Error('No user is logged in.'));
-      }
       const promotions = [...uploadedPromotions];
       const promotion = promotions.find(({ id }) => id === promotionId);
       if (promotion) {
         if (promotion.isSavedByUser) {
           promotion.isSavedByUser = false;
-          return UserService.unsavePromotion(authUser.user.id, promotionId)
+          return UserService.unsavePromotion(authUser?.user.id || '', promotionId)
             .then(() => setUploadedPromotions(promotions))
             .catch(() => null);
         }
         promotion.isSavedByUser = true;
-        return UserService.savePromotion(authUser.user.id, promotionId)
+        return UserService.savePromotion(authUser?.user.id || '', promotionId)
           .then(() => setUploadedPromotions(promotions))
           .catch(() => null);
       }
     },
-    [uploadedPromotions, setUploadedPromotions, authUser]
+    [authUser, uploadedPromotions, setUploadedPromotions]
+  );
+
+  /**
+   * If the user has not downvoted the promotion, downvote the promotion. Otherwise, set back to initial state.
+   *
+   * @param promotionId - The id of the promotion to downvote
+   */
+  const onDownvoteClick = useCallback(
+    (promotionId: string): void => {
+      const promos = [...uploadedPromotions];
+      const promotion = promos.find(({ id }) => id === promotionId);
+      if (promotion) {
+        if (promotion.voteState === VoteState.DOWN) {
+          promotion.votes = promotion.votes + 1;
+          promotion.voteState = VoteState.INIT;
+          PromotionService.downvotePromotion(promotionId, authUser?.user.id || '')
+            .then(() => setUploadedPromotions(promos))
+            .catch(() => null);
+        } else {
+          if (promotion.voteState === VoteState.UP) {
+            // Remove upvote
+            promotion.votes = promotion.votes - 1;
+          }
+          promotion.votes = promotion.votes - 1;
+          promotion.voteState = VoteState.DOWN;
+          PromotionService.downvotePromotion(promotionId, authUser?.user.id || '')
+            .then(() => setUploadedPromotions(promos))
+            .catch(() => null);
+        }
+      }
+    },
+    [authUser, uploadedPromotions, setUploadedPromotions]
+  );
```

Comment:
Same comment for these ones too about empty string for userId and remove duplication